custom/icds_reports/static/js/angular-services/baseControllers.service.js

```diff
@@ -233,6 +233,29 @@ window.angular.module('icdsApp').factory('baseControllersService', function() {
                 }
             };
 
+            //Creating a promise, which resolves only after map is rendered (wait until canvas is rendered in dom)
+            // Reference: https://stackoverflow.com/a/47776379/12839195
+            vm.rafAsync = function () {
+                return new Promise(function (resolve) {
+                    requestAnimationFrame(resolve);
+                });
+            };
+
+            vm.checkElement = function (selector) {
+                // if it is not a page where map is displayed, resolve the promise immediately,
+                // else wait till canvas (svg in dom) is created
+                if (!navigationService.isMapDisplayed($location.path())) {
+                    return Promise.resolve(true);
+                }
+                if (document.querySelector(selector) === null) {
+                    return vm.rafAsync().then(() => vm.checkElement(selector));
+                } else {
+                    return Promise.resolve(true);
```

Comment:
Guessing this is the root cause of the test failure. The version of javascript the tests run doesn't have promises so I think you have to add a polyfill.

I started down this path here: https://github.com/dimagi/commcare-hq/commit/6bf39a8697fcad1e79b0e7c247655d6e32b2df6d but gave up because there were other issues...